it/controllers/SABlueInterruptControllerISpec.scala

```diff
@@ -18,33 +18,404 @@ package controllers
 
   s"GET $urlPath" when {
-    "a user has an SA Enrolment associated to the credentials on another account" should {
-      s"return $OK with the SABlueInterrupt page" in {
+    "the session cache has a credential for SA enrolment that is not the signed in account" should {
+      s"render the report blue interupt page" in {
+        await(save[String](sessionId, "redirectURL", returnUrl))
+        await(
+          save[AccountTypes.Value](
+            sessionId,
+            "ACCOUNT_TYPE",
+            SA_ASSIGNED_TO_OTHER_USER
+          )
+        )
         val authResponse = authoriseResponseJson()
         stubAuthorizePost(OK, authResponse.toString())
         stubPost(s"/write/.*", OK, """{"x":2}""")
-
-        val res = buildRequest(urlPath,followRedirects = true)
-          .withHttpHeaders(xSessionId, csrfContent)
+        val res = buildRequest(urlPath, followRedirects = false)
+          .withHttpHeaders(xSessionId, xRequestId, sessionCookie)
           .get()
 
         whenReady(res) { resp =>
+          val page = Jsoup.parse(resp.body)
+
           resp.status shouldBe OK
-          resp.body should include(
+          page.title should include(
             "Before you continue to your personal tax account"
           )
         }
       }
     }
+
+    List(
+      SINGLE_ACCOUNT,
+      PT_ASSIGNED_TO_CURRENT_USER,
+      MULTIPLE_ACCOUNTS,
+      PT_ASSIGNED_TO_OTHER_USER,
+      SA_ASSIGNED_TO_CURRENT_USER
+    ).foreach { accountType =>
+      s"the session cache has a credential with account type ${accountType.toString}" should {
+        s"redirect to account check page" in {
+          await(save[String](sessionId, "redirectURL", returnUrl))
+          await(
+            save[AccountTypes.Value](sessionId, "ACCOUNT_TYPE", accountType)
+          )
+          val authResponse = authoriseResponseJson()
+          stubAuthorizePost(OK, authResponse.toString())
+          stubPost(s"/write/.*", OK, """{"x":2}""")
+          val res = buildRequest(urlPath, followRedirects = false)
+            .withHttpHeaders(xSessionId, xRequestId, sessionCookie)
+            .get()
+
+          whenReady(res) { resp =>
+            resp.status shouldBe SEE_OTHER
+            resp.header("Location").get should include(s"/protect-tax-info")
+          }
+        }
+      }
+    }
+
+    "the session cache has no redirectUrl" should {
+      "render the error page" in {
+        val authResponse = authoriseResponseJson()
+        stubAuthorizePost(OK, authResponse.toString())
+        stubPost(s"/write/.*", OK, """{"x":2}""")
+        val res = buildRequest(urlPath, followRedirects = false)
+          .withHttpHeaders(xSessionId, xRequestId, sessionCookie)
+          .get()
+
+        whenReady(res) { resp =>
+          resp.status shouldBe OK
+          resp.body should include("There was a problem")
+        }
+      }
+    }
+
+    "an authorised user with no credential uses the service" should {
```

Comment:
am I right in thinking alot of these tests below are generic, I added a comment in the other ISPEC you added, perhaps we should split these out and make a helper? if its possible